### ISSUE TITLE
Permission denied error in setup script on RBP3

### ISSUE THREAD

**Issue by ubx-caloca** [openthread/borderrouter#133]
I'm following this guide (https://openthread.io/guides/border_router/build) to get and test the OTBR. However, I'm getting an error "mkdir: cannot create directory /etc/ncp_state_notifier: Permission denied"  every time I execute "./script/setup".

It seems that the script is trying to do an instruction that needs root permission. I'm I suppose to run the setup script with sudo?.

I've attached the output log of the setup script.  
[setup.log](https://github.com/openthread/borderrouter/files/1679378/setup.log)

Details of my working environment are shown below:

- OTBR commit: Jan 17 20:29:59 2018 - Improve DHCPv6 PD. (#128) 
- Using a fresh Raspbian Stretch Lite 2017-11-29 installation

**Comment by jwhui** [comment]
I just merged #131, which should help address your issue.  Can you sync with the latest master and try again?

**Comment by ubx-caloca** [comment]
Thanks for the quick reply. Just finished running the setup script and the error is no more.  I will continue to follow the Openthread.io guide to see if everything is installed correctly, I was having issues with the Commissioning app not seeing the BR.

**Comment by ubx-caloca** [comment]
Mmm it seems that setup script didn't work as expected. When executing the script after a while the Raspberry restarts on its own, I thought that it was the expected behavior after finishing, however when verifying services with command "sudo systemctl status" I'm not seeing otbr-agent.service, otbr-web.service and wpantund.service.

Does the setup script generates some log files to check what is happening?

**Comment by bukepo** [comment]
@ubx-caloca I think you may save log by redirect stdout and stderr. What version of Raspberry are you using? In some low memory boards, you have to increase the swap to finish compiling the project. See #115.

**Comment by ubx-caloca** [comment]
I'm using a Raspberry Pi 3 Model B and installed a fresh Raspbian Stretch Lite 2017-11-29.  I will send you the log file tomorrow and try to increase the swap file.

**Comment by ubx-caloca** [comment]
I'm sending the logs of the setup script generating by typing "./scripts/setup &> setupOutput.log".  

[setupOutput.log](https://github.com/openthread/borderrouter/files/1687082/setupOutput.log)

After a while of executing the script, the raspberry rebooted by itself. In addition, the otbr services were not installed, here is the output of the "sudo systemctl status" command.

[systemctlStatus.txt](https://github.com/openthread/borderrouter/files/1687095/systemctlStatus.txt)

If you need more info about my environment, please let me know. I will try enabling the swapfile.

**Comment by ubx-caloca** [comment]
Found the reboot issue of the setup script. I was using a power adapter of one of my electronic devices because I was giving the RBPi3 without the power adapter. It was working fine with the Pi, however, it seems that when it was in the middle of compiling stuff the power setup was not enough to power the RBPi3 and cause the reboot. 

For future reference use a power adapter with 5V and 2A, mine was a 5V 850mA and was causing the reboot.

I've moved along with the OTBR installation and found some other unrelated issues, I will post them in a separate Github issue. As for the original permission denied issue, the new commit indeed fixes it.
